### PR TITLE
chore: Upgrade UI

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
* Update node js from 16 to 20
   * Add engines.node field in package.json across packages to pin this `engines.node >=20.13.1`
   * Update GitHub CI build-ui job to use node 20
* Update React from 17 to 18
* Update dependencies. Notable change: elastic ui
* Update dev dependencies. Notable change: Craco and Rollup
* Update GitHub CI jobs to use latest version of actions

### CODE REVIEW

File: ui/packages/app/src/PrivateLayout.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   ApplicationsContextProvider,
   Header,
   PrivateRoute,
   ProjectsContextProvider
 } from "@caraml-dev/ui-lib";
-import config from "./config";
 import urlJoin from "proper-url-join";
+import React from "react";
 import { Outlet, useNavigate } from "react-router-dom";
+import config from "./config";
```
Comment (leonlnj) [inline ui/packages/app/src/PrivateLayout.js:11]: why did the imports get re-sorted?

File: ui/packages/lib/src/index.scss [hunk 2]
```diff
@@ -2,19 +2,13 @@
...
   padding-top: 0 !important;
 }
 
 #root {
   display: flex;
   flex-direction: column;
   min-height: 100vh;
-
-  .euiBody--headerIsFixed & {
-    min-height: calc(100vh - #{$euiHeaderHeightCompensation});
```
Comment (leonlnj) [inline ui/packages/lib/src/index.scss:16]: was this actually unused all the while

File: ui/packages/app/src/PrivateLayout.js (hunk 1 above)
Comment (ariefrahmansyah) [reply ui/packages/app/src/PrivateLayout.js:11]: Most likely because of my IDE's auto formatter.

File: ui/packages/lib/src/index.scss (hunk 2 above)
Comment (ariefrahmansyah) [reply ui/packages/lib/src/index.scss:16]: It was, but new update of EUI deprecate it.

**Review by leonlnj — APPROVED** [review]
thanks! lgtm

**Pull request merged**